### PR TITLE
[MCU change] Update Infinity87 rev. 2

### PR DESCRIPTION
**Pull request by mechlovin**
MCU changed from Atmega32A to Atmega32u4

## Description

 This design originally used Atmega32A as a budget-conscious choice due to the microcontroller shortage experienced recently. However, it lacks certain features that are necessary for our current TKL lineup, so we decided to switch to Atmega32u4 with minimal changes regarding the matrix pins and LED pins. Thank you for your support!

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Rebase tends to work better than merge, but isn't as easy, in a number of cases. 

if you would like I can see about rebasing this, so that it doesn't have the extra stuff. 
(the caveat is that it requires some extra work on your end, still, to properly pull the updated branch)

**Comment by mechlovin** [comment]
> Rebase tends to work better than merge, but isn't as easy, in a number of cases.
> 
> if you would like I can see about rebasing this, so that it doesn't have the extra stuff. (the caveat is that it requires some extra work on your end, still, to properly pull the updated branch)

Thank you very much for your suggestions. We conducted a rebase following this instruction: https://docs.qmk.fm/#/newbs_git_resolving_merge_conflicts?id=rebasing-your-changes. On top of that,  we would really appreciate your further guidance and wouldn't mind the extra work. Whatever it takes to have this done properly.

**Comment by mechlovin** [comment]
I have run the commands as you instructed. 
image

Do I have to attend to this: 
image
 or just wait for further approval? 

Thank you! @drashna

**Comment by drashna** [comment]
Should be fine waiting for further approval. 

But the reset didn't complete correctly.   you want `origin` rather than the URL in there.

File: keyboards/mechlovin/infinity87/rev2/readme.md
Comment (fauxpark) [inline keyboards/mechlovin/infinity87/rev2/readme.md:16]:
```suggestion
    make mechlovin/infinity87/rev2:flash
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Are there any 32A versions of this board already out in the wild? If so, this needs to be a separate board, not a change to the existing one, so that users with the "old" version will still be able to build keymaps.

File: keyboards/mechlovin/infinity87/rev2/readme.md
Comment (mechlovin) [reply keyboards/mechlovin/infinity87/rev2/readme.md:16]: Hi @fauxpark 

The 32A version only made it to a working prototype unit that's currently in our possession, and has never been offered to the public. 

We decided to use 32A on several new designs amidst the global chip shortage, but as new projects come in, we reconsidered our plan and decided to use 32u4 across our TKL lineup. Our apologies for the confusion. We have been taking notes to improve our practice so inconsistencies like this can be avoided in the future. 

Thank you!

**Review by fauxpark — APPROVED** [review]

**Pull request merged**
